lib/paths/index.ts

```diff
@@ -54,7 +63,7 @@ type RecursivePaths<
       ValueOf<{
         [Key in keyof Type]: Key extends Pathable
           ?
-              | `${Key}`
+              | `${AnyArrayIndexAccessorOrKey<Key, CallOptions>}`
```

Comment:
Since you don't update `anyArrayIndexAccessor` in `CallOptions` (unlike `depth`), I suggest using `UserOptions`:

```suggestion
              | `${AnyArrayIndexAccessorOrKey<Key, UserOptions>}`
```